File: 基礎題庫/a006.cpp
#include <iostream>
#include <cmath>

using namespace std ;

int main () {

	int a, b, c ;
	while ( cin >> a >> b >> c ) {

		int D = ( pow ( b, 2 ) - 4 * a * c ) ;
		int x1 = ( -b + sqrt ( D ) ) / ( 2 * a ), x2 = ( -b - sqrt ( D ) ) / ( 2 * a ) ;

		if ( D < 0 )
			cout << "No real root" << endl ;

		else if ( D == 0 )
			cout << "Two same roots x=" << x1 << endl ;

		else
			cout << "Two different roots x1=" << x1 << " , x2=" << x2 << endl ;
	}

	return 0 ;

}

File: 基礎題庫/Python/a006.py
import sys

for sInput in sys.stdin:
    a, b, c = map(int, sInput.split())
    jdg = b ** 2 - 4 * a * c

    if jdg < 0:
        print("No real root");
    elif jdg == 0:
        ans = int((-b + jdg ** 0.5) / (2 * a))
        print("Two same roots x=" + str(ans))
    else:
        ans1 = int((-b + jdg ** 0.5) / (2 * a))
        ans2 = int((-b - jdg ** 0.5) / (2 * a))
        print("Two different roots x1=" + str(ans1) + " , x2=" + str(ans2))